File: API.NEWS/Controllers/NewsController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Mvc;

namespace API.NEWS.Controllers
{
    [ApiController]
    [Route("api/[controller]")]
    public class NewsController : ControllerBase
    {
        private readonly ApplicationDbContext _dbContext;
        public NewsController (ApplicationDbContext dbContext) {
            _dbContext = dbContext;
        }
        
        [HttpGet]
        [Authorize]
        [Route("show-news")]
        public async Task<IActionResult> showNews() {
            var news = await _dbContext.News.ToListAsync();
            if (news.Count  == 0) {
                return BadRequest(new {
                status = false,
                message = "news is empty, create data first. ",
                data = new {}
                });
            }
            return Ok(new {
                status = true,
                message = "this news list.",
                data = news
                });
        }

        [HttpGet]
        [Authorize]
        [Route("detail-news")]
        public async Task<IActionResult> detailNews(int id) {
            var news = await _dbContext.News.FindAsync(id);
            if(news == null) {
                return BadRequest(new {
                status = false,
                message = $"news dengan id {id} tidak ada",
                data = new {}
                });
            }
            return Ok(new {
                status = true,
                message = $"news detail {id}",
                data = news
                });
        }

        [HttpPost]
        [Authorize]
        [Route("create-news")]
        public async Task<IActionResult> createNews([FromForm] NewsDto create) {
            var uploadImage = Path.Combine(Directory.GetCurrentDirectory(), "wwwroot", "Image");
            if(!Directory.Exists(uploadImage))
                Directory.CreateDirectory(uploadImage);
            
            var fileImage = $"{create.Title}-{create.FileImage.FileName}";
            var filePath = Path.Combine(uploadImage, fileImage);

            using var stream = System.IO.File.Create(filePath);
            if(create.FileImage != null)
                create.FileImage.CopyTo(stream);
            
            var url = $"{Request.Scheme}://{Request.Host}{Request.PathBase}/Image/{fileImage}";
             var newNews = new News
            {
                Title = create.Title,
                FileName = fileImage,
                Url = url,
                Content = create.Content,
                PublishDate = (DateTime)create.PublishDate,
            };
                _dbContext.Add(newNews);
                await _dbContext.SaveChangesAsync();
            return Ok(new {
                status = true,
                message = "news has been created !",
                data = newNews
                });
        }

        [HttpPost]
        [Authorize]
        [Route("update-news")]
        public async Task<IActionResult> UpdateNews(int id, [FromForm] NewsDto update)
        {
            var uploadImage = Path.Combine(Directory.GetCurrentDirectory(), "wwwroot", "Image");
            if (!Directory.Exists(uploadImage))
                Directory.CreateDirectory(uploadImage);

            var news = await _dbContext.News.FindAsync(id);
            if (news == null)
            {
                return NotFound();
            }

            if (update.FileImage == null)
            {
                // Jika input file kosong, gunakan data yang sudah ada sebelumnya
                news.Title = update.Title;
                news.Content = update.Content;
                news.PublishDate = (DateTime)update.PublishDate;
            }
            else
            {
                // Jika input file tidak kosong, hapus file lama dan gunakan data baru
                var fileImage = $"{update.Title}-{update.FileImage.FileName}";
                var filePath = Path.Combine(uploadImage, fileImage);

                using var stream = System.IO.File.Create(filePath);
                update.FileImage.CopyTo(stream);

                var url = $"{Request.Scheme}://{Request.Host}{Request.PathBase}/Image/{fileImage}";

                // Hapus file lama
                var oldFilePath = Path.Combine(uploadImage, news.FileName);
                if (System.IO.File.Exists(oldFilePath))
                    System.IO.File.Delete(oldFilePath);

                news.Title = update.Title;
                news.FileName = fileImage;
                news.Url = url;
                news.Content = update.Content;
                news.PublishDate = (DateTime)update.PublishDate;
            }

            _dbContext.News.Update(news);
            await _dbContext.SaveChangesAsync();
            return Ok(new
            {
                status = true,
                message = "News has been updated!",
                data = news
            });
        }


        [HttpPost]
        [Authorize]
        [Route("delete-news")]
        public async Task<IActionResult> DeleteNews (int Id) {
            var news = _dbContext.News.First( x => x.Id == Id);

            _dbContext.News.Remove(news);
            await _dbContext.SaveChangesAsync();
            return Ok(new {
                status = true,
                message = "news has been deleted !",
                data = new {}
                });
        }
    }
}

File: app/Http/Controllers/NewsController.php
<?php

namespace App\Http\Controllers;
use App\Helpers\HttpClient;
use Illuminate\Http\UploadedFile;
use Illuminate\Http\Request;

class NewsController extends Controller
{
    //
    public function showNews()
    {
        $getNews = HttpClient::fetch("GET", HttpClient::apiUrl()."News/"."show-news");
        $news = $getNews['data'];
        if ($getNews['status'] == false) {
            return redirect('/create-news')->with('error', $getNews['message']);
        } else if(isset($getNews['errors'])) {
            $errors = $getNews['errors'];
            $message = array_values($errors)[0][0];
            return redirect('/create-news')->with('error', $message);
        }

        return view('shownews', compact('news'));
    }

    public function detailNews($id)
    {
        $getNews = HttpClient::fetch(
        "GET",
         HttpClient::apiUrl()."News/detail-news?id=".$id,
        );
        $news = $getNews['data'];
        if ($getNews['status']==false) {
        return redirect('/show-news')->with('error', $getNews['message']);
        }else if(isset($getNews['errors'])) {
            $errors = $getNews['errors'];
            $message = array_values($errors)[0][0];
            return redirect('/show-news')->with('error', $message);
        }
        return view('detailnews', compact('news'));
    }

    public function createNews(Request $request)
    {
        //dd(Carbon::parse($request->PublishDate)->utc());
        $date = date('Y-m-d\TH:i:sp');
        $file = [
            'FileImage' => $request->file('FileImage')
        ];
        $payload = [
            'Title' => $request->Title,
            'Content' => $request->Content,
            'PublishDate' => $date,
        ];

        $create = HttpClient::fetch(
            "POST",
            HttpClient::apiUrl()."News/"."create-news",
            $payload,
            $file,
        );

        if ($create['status'] == false) {
        return redirect('/create-news')->with('error', $create['message']);
        } else if(isset($create['errors'])) {
            $errors = $create['errors'];
            $message = array_values($errors)[0][0];
            return redirect('/create-news')->with('error', $message);
        }

        return redirect('/show-news')->with('success', $create['message']);
    }

    public function editNews($id)
    {
        $getNews = HttpClient::fetch(
        "GET",
         HttpClient::apiUrl()."News/detail-news?id=".$id,
        );
        $news = $getNews['data'];
        if ($getNews['status']==false) {
        return redirect('/show-news')->with('error', $getNews['message']);
        }else if(isset($getNews['errors'])) {
            $errors = $getNews['errors'];
            $message = array_values($errors)[0][0];
            return redirect('/show-news')->with('error', $message);
        }
        return view('updatenews', compact('news'));
    }
    
    public function updateNews(Request $request, $id)
{
    $date = date('Y-m-d\TH:i:sp');
    $file=[];
    $payload = [
        'Title' => $request->Title,
        'Content' => $request->Content,
        'PublishDate' => $date,
    ];
    if ($request->hasFile('FileImage')) {
        // Jika input file tidak kosong, gunakan data baru
        $file = [
            'FileImage' => $request->file('FileImage')
        ];
    } 


    $update = HttpClient::fetch(
        "POST",
        HttpClient::apiUrl()."News/"."update-news?id=".$id,
        $payload,
            $file,
    );


    if ($update['status'] == false) {
        return redirect('/show-news')->with('error', $update['message']);
    } else if(isset($update['errors'])) {
        $errors = $update['errors'];
        $message = array_values($errors)[0][0];
        return redirect('/show-news')->with('error', $message);
    }
    
    return redirect('/show-news')->with('success', $update['message']);
}


    public function destroy($id)
    {
        $delete = HttpClient::fetch(
            "POST",
            HttpClient::apiUrl()."News/delete-news?id=".$id,
        );
        if ($delete['status'] == false) {
            return redirect('/show-news')->with('error', $delete['message']);
        } else if(isset($delete['errors'])) {
            $errors = $delete['errors'];
            $message = array_values($errors)[0][0];
            return redirect('/show-news')->with('error', $message);
        }
        return redirect('/show-news')->with('success', $delete['message']);
    }
}
